GitHubCodespaces.ps1

```diff
@@ -0,0 +1,538 @@
+#>
+    [CmdletBinding(DefaultParameterSetName = 'AuthenticatedUser')]
+    [OutputType({$script:GitHubCodespaceTypeName})]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "", Justification="The Uri parameter is only referenced by Resolve-RepositoryElements which get access to it from the stack via Get-Variable -Scope 1.")]
+    param(
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Repository')]
+        [string] $OwnerName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Repository')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='CodespaceName')]
+        [string] $CodespaceName,
+
+        [Parameter(
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Organization')]
+        [string] $OrganizationName,
+
+        [Parameter(
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Organization')]
+        [ValidateNotNullOrEmpty()]
+        [String] $UserName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        'UsageType' = $PSCmdlet.ParameterSetName
+    }
+
+    $uriFragment = [String]::Empty
+    $description = [String]::Empty
+    switch ($PSCmdlet.ParameterSetName)
+    {
+        'AuthenticatedUser' {
+            # /user/codespaces
+            $uriFragment = 'user/codespaces'
+            $description = 'Getting codespaces for current authenticated user'
+
+            break
+        }
+
+        'CodespaceName' {
+            $telemetryProperties['CodespaceName'] = Get-PiiSafeString -PlainText $CodespaceName
+
+            $uriFragment = "user/codespaces/$CodespaceName"
+            $description = "Getting user/codespaces/$CodespaceName"
+
+            break
+        }
+
+        'Organization' {
+            # /orgs/{org}/codespaces
+            # /orgs/{org}/members/{username}/codespaces
+
+            $telemetryProperties['OrganizationName'] = Get-PiiSafeString -PlainText $OrganizationName
+            if ([string]::IsNullOrWhiteSpace($UserName)) {
+                $uriFragment = "orgs/$OrganizationName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+            else {
+                $telemetryProperties['UserName'] = Get-PiiSafeString -PlainText $UserName
+                $uriFragment = "orgs/$OrganizationName/members/$UserName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+
+            break
+        }
+
+        'Repository' {
```

Comment:
Your `Repository` and `Uri` cases are identical, which leads to code duplication.
Please collapse those into a single case statement. You can write it like this:

```powershell
         { $_ -in ('Repository', 'Uri') } {
                 # ...
         }
```